integration-tests/test_satellite.py

```diff
@@ -0,0 +1,113 @@
+    previous_hostname = output.decode("utf-8").strip()
+
+    try:
+        # Set a new hostname
+        hostname_to_test = "insights-client-satellite"
+        set_hostname(hostname_to_test)
+        cmd = ["hostnamectl", "status"]
+        output = subprocess.check_output(cmd)
+        assert hostname_to_test in output.decode("utf-8").strip()
+
+        # redaction on hostname:
+        data_redaction(hostname_to_test)
+
+        # check the collection, if redaction pattern is filtered out
+        # collect data
+        archive_result = insights_client.run("--no-upload")
+        assert archive_result.returncode == 0
+
+        # extract the collected tar file
+        archive_location = archive_result.stdout.split("\n")[-2].split(" ")[-1]
+        cmd = ["tar", "-xzvf", archive_location]
+        subprocess.check_call(cmd)
+
+        # the redaction data should not locate in the collected files
+        archive_extract = archive_location.split(".")[0].split("/")[-1]
+        cmd = "cd %s; grep -rnw . -e %s" % (archive_extract, hostname_to_test)
+        # should not find the obfuscate pattern, the return code should not zero.
+        returncode = subprocess.check_call(cmd, shell=True)
+        assert returncode
+    except Exception as e:
+        traceback.print_exc()
+    finally:
+        # Recover the previous hostname for the system
+        set_hostname(previous_hostname)
+        # Remove all data redaction
+        cmd = ["rm", "-f", "/etc/insights-client/file-content-redaction.yaml"]
+        subprocess.check_call(cmd)
+
+
+def test_set_ansible_host_info(insights_client):
+    """
+    Test if the ansible-host can be set with satellite
+    Related to https://issues.redhat.com/browse/RHELPLAN-160348,
+    and https://bugzilla.redhat.com/show_bug.cgi?id=222729
+    """
+    # Register system against Satellite, and register insights through satellite
+    insights_client.register()
+    sleep(10)
+    assert insights_client.is_registered
+
+    # Update ansible-host
+    set_ansible_host = insights_client.run("--ansible-host='foo.example.com'")
+    assert set_ansible_host.returncode == 0
+
+
+def test_register_group_option(insights_client):
+    """
+    Test registering host when using --group option through satellite
+    Related to https://bugzilla.redhat.com/show_bug.cgi?id=2059031,
+    and https://issues.redhat.com/browse/RHEL-3739
+    """
+    # make sure the system is not registered to insights
+    with contextlib.suppress(Exception):
+        insights_client.unregister()
+    assert not insights_client.is_registered
+    """ Register the system to insights with group option
+    Use insights_client.run can not find the existing bug, so try subprocess
+    register_with_group = insights_client.run("--register", "--group='tag'")
+    assert register_with_group.returncode == 0
+    """
+    cmd = ["insights-client", "--register", "--group='tag'"]
+    subprocess.check_call(cmd, shell=True)
```

Comment:
I forgot the details with the invocation after a weekend... Let me try when I'm free.

Reply:
------------------------------------------------------------- live log call -------------------------------------------------------------
DEBUG    pytest_client_tools:util.py:87 running ['insights-client', '--unregister'] with options {'check': True, 'stdout': -1, 'stderr': -1, 'text': True}
DEBUG    pytest_client_tools:util.py:95 result: CompletedProcess(args=['insights-client', '--unregister'], returncode=1, stdout='This host is not registered, unregistration is not applicable.\n', stderr='')
DEBUG    pytest_client_tools:util.py:87 running ['insights-client', '--status'] with options {'check': False, 'stdout': -1, 'stderr': -1, 'text': True}
DEBUG    pytest_client_tools:util.py:95 result: CompletedProcess(args=['insights-client', '--status'], returncode=0, stdout='This host is unregistered.\n', stderr='')
DEBUG    pytest_client_tools:util.py:87 running ['insights-client', '--register', "--group='tag'"] with options {'check': True, 'stdout': -1, 'stderr': -1, 'text': True}
DEBUG    pytest_client_tools:util.py:95 result: CompletedProcess(args=['insights-client', '--register', "--group='tag'"], returncode=0, stdout='Automatic scheduling for Insights has been enabled.\nStarting to collect Insights data for dell-per740-68-vm-03.lab.eng.pek2.redhat.com\nWriting RHSM facts to /etc/rhsm/facts/insights-client.facts ...\nUploading Insights data.\nSuccessfully uploaded report for dell-per740-68-vm-03.lab.eng.pek2.redhat.com.\nView the Red Hat Insights console at https://console.redhat.com/insights/\n', stderr='')
PASSED

Reply:
When I run "insights_client.run("--register", "--group='tag'")", I got PASS result.

Reply:
> When I run "insights_client.run("--register", "--group='tag'")", I got PASS result.

that sounds like it should be good to use... or am I missing anything?

Reply:
Bug https://issues.redhat.com/browse/RHINENG-7567 is not fixed yet, and I can reproduce this bug manually, and I can reproduce this bug by subprocess way to run insights-client. I don't know why "insights_client.run("--register", "--group='tag'")" does not reproduce this bug.

Reply:
I have a potential idea: what if you try:
```python
insights_client.run("--register", "--group=tag")
```
i.e. without single quotes around the value of `--group`.

I'm thinking that, looking at the code:
```python
    cmd = ["insights-client", "--register", "--group='tag'"]
    subprocess.check_call(cmd, shell=True)
```
since this uses a shell, in the end it is like invoking `insights-client --register --group='tag'`, and the quoting of `tag` is eaten by the shell, so `--group` only gets `tag` without quotes. When using `run()` the shell is not used, so `--group` gets a literal `'tag'` (with quotes) as value.

LMK (Let Me Know) :)

Reply:
The following is what if I use insights_client.run("--register", "--group=tag"):


------------------------------------------------------------- live log call -------------------------------------------------------------
DEBUG    pytest_client_tools:util.py:87 running ['insights-client', '--unregister'] with options {'check': True, 'stdout': -1, 'stderr': -1, 'text': True}
DEBUG    pytest_client_tools:util.py:95 result: CompletedProcess(args=['insights-client', '--unregister'], returncode=1, stdout='This host is not registered, unregistration is not applicable.\n', stderr='')
DEBUG    pytest_client_tools:util.py:87 running ['insights-client', '--status'] with options {'check': False, 'stdout': -1, 'stderr': -1, 'text': True}
DEBUG    pytest_client_tools:util.py:95 result: CompletedProcess(args=['insights-client', '--status'], returncode=0, stdout='This host is unregistered.\n', stderr='')
DEBUG    pytest_client_tools:util.py:87 running ['insights-client', '--register', '--group=tag'] with options {'check': True, 'stdout': -1, 'stderr': -1, 'text': True}
DEBUG    pytest_client_tools:util.py:95 result: CompletedProcess(args=['insights-client', '--register', '--group=tag'], returncode=0, stdout='Automatic scheduling for Insights has been enabled.\nStarting to collect Insights data for dell-per740-68-vm-03.lab.eng.pek2.redhat.com\nWriting RHSM facts to /etc/rhsm/facts/insights-client.facts ...\nUploading Insights data.\nSuccessfully uploaded report for dell-per740-68-vm-03.lab.eng.pek2.redhat.com.\nView the Red Hat Insights console at https://console.redhat.com/insights/\n', stderr='')
PASSED

Reply:
I tried to use insights_client.run("--register", "--group=tag") and set insights.legacy_upload = true in .secret.tomal. Yes, this case fails.

But bug https://issues.redhat.com/browse/RHINENG-7567 is not fixed when insights-client(legacy_upload=true) integrates with satellite. I prefer to add a check for the satellite env.

Reply:
I tried manually, and found:
1. Bug https://issues.redhat.com/browse/RHINENG-7567 exists on both  production env and satellite env with legacy_upload=True.
2. When configuring legacy_upload=False, "insights-client --register --group=tag" works well on both envs.